server/test/unit/server/webservices/views/test_task_groups.py

```diff
@@ -27,6 +28,55 @@ def filter(self, state=None):
         return MockQuerySet(filtered_list)
 
 
+class TestTaskGroupView(unittest.TestCase):
+    """
+    Tests for TaskGroupView
+    """
+    @mock.patch('pulp.server.webservices.views.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.task_groups.TaskStatus.objects')
+    @mock.patch('pulp.server.webservices.views.task_groups.tasks')
+    def test_delete_task_resource_nonexistant(self, mock_task, mock_objects):
+        """
+        Test delete task_group with no tasks
+        """
+        mock_request = mock.MagicMock()
+        mock_objects.only.return_value.get_or_404.side_effect = MissingResource
+        task_resource = TaskGroupView()
+        try:
```

Comment:
It would be better to just use ``self.assertRaises`` here. Don't worry about testing the response code on the exception, because that's a property of the exception itself. That should be tested elsewhere.